src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -85,7 +85,10 @@ protected ApiConfiguration Configuration
                     apiType =>
                     {
                         IServiceCollection services = new ServiceCollection()
-                            .CutoffPrevious<IQueryExecutor>(DefaultQueryExecutor.Instance);
+                            .CutoffPrevious<IQueryExecutor>(DefaultQueryExecutor.Instance)
+                            .AddSingleton<ApiConfigurationExtensions.PropertyBag>()
+                            .AddScoped<ApiContextExtensions.PropertyBag>()
+                            .AddScoped<InvocationContextExtensions.PropertyBag>();
```

Comment:
Exactly. But then I found it hard to determine the best place to put the "scoped" `PropertyBag`. An alternative would be to put the two `PropertyBag`s (one for `ApiConfiguration` and one for scope) into `PropertyBagBase.cs`. But to keep consistent, I bind each one to its own extension class because the `PropertyBag` logic are only offered through the extension classes. Do you have any suggestion?